x-pack/plugins/security_solution/public/timelines/components/fields_browser/index.test.tsx

```diff
@@ -34,14 +36,6 @@ const defaultDataviewState: {
   missingPatterns: [],
   selectedDataViewId: 'security-solution',
 };
-const mockScopeIdSelector = jest.fn(() => defaultDataviewState);
```

Comment:
this is weird. Not sure why anything I did in this PR would break this test, but it did. Before these changes, this test gave the following error:

```
FAIL  x-pack/plugins/security_solution/public/timelines/components/fields_browser/index.test.tsx
--
  | ● Test suite failed to run
  |  
  | TypeError: Cannot read properties of undefined (reading 'createStore')
  |  
  | 13 | import { createStoreFactory, createStore, getStore } from './store';
  | 14 | import type { ImmutableMiddleware, State } from './types';
  | > 15 | import type { AppAction } from './actions';
  | |                   ^
  | 16 | import type { Immutable } from '../../../common/endpoint/types';
  | 17 |
  | 18 | export { createStoreFactory, createStore, getStore };
  |  
  | at Object.get [as createStore] (x-pack/plugins/security_solution/public/common/store/index.ts:15:19)
  | at x-pack/plugins/security_solution/public/timelines/components/fields_browser/index.test.tsx:40:3
  | at Object.require (x-pack/plugins/security_solution/public/detections/components/alerts_table/grouping_settings/group_take_action_items.tsx:13:1)
  | at Object.require (x-pack/plugins/security_solution/public/detections/components/alerts_table/grouping_settings/index.tsx:13:1)
  | at Object.require (x-pack/plugins/security_solution/public/common/store/grouping/reducer.ts:9:1)
  | at Object.require (x-pack/plugins/security_solution/public/common/store/reducer.ts:32:1)
  | at Object.require (x-pack/plugins/security_solution/public/common/store/index.ts:9:1)
  | at requireActual (x-pack/plugins/security_solution/public/timelines/components/fields_browser/index.test.tsx:39:25)
  | at Object.require (x-pack/plugins/security_solution/public/timelines/components/fields_browser/index.tsx:19:1)
  | at Object.require (x-pack/plugins/security_solution/public/timelines/components/fields_browser/index.test.tsx:11:1)
```

I changed up the mocks to fix it. @semd, I see you wrote most of this test file. Any idea why it would be upset about my changes?

Reply:
Yes, this is weird, the error is not pointing to the correct line. 

But the root cause of this error seems to be a circular dependency, by looking at the error trace the `jest.requireActual('../../../common/store')` ends up importing itself in the _group_take_action_items.tsx_.

The problem is in the _public/common/store/grouping/reducer.ts_ that is importing from _public/detections/components/alerts_table/grouping_settings/index.tsx_. Modules inside _common_ should never import stuff from _components_, the dependency flow should work only the other way around (_components_ load _common_) otherwise it will very likely create a circular dependency.

We'll have to move the `getDefaultGroupingOptions` to the _public/common/utils/alerts.ts_ of create an _alert_grouping_ dir or something inside _common_, and then change the everywhere.

I think there's a linter rule for that, I'll investigate.